device/thunder_ripple_sdk/src/bootstrap/boot_thunder.rs

```diff
@@ -21,25 +21,105 @@ use crate::{
 };
 use ripple_sdk::{
     extn::client::extn_client::ExtnClient,
-    log::{error, info},
+    log::{debug, error, info, warn},
+    serde_json::{self, Error},
 };
 
 use super::{get_config_step::ThunderGetConfigStep, setup_thunder_pool_step::ThunderPoolStep};
+use crate::client::thunder_client::ThunderClientBuilder;
+use crate::thunder_state::ThunderBootstrapStateWithConfig;
+use crate::thunder_state::ThunderState;
+use ripple_sdk::api::config::Config;
+use ripple_sdk::utils::error::RippleError;
+use serde::Deserialize;
+
+use ripple_sdk::extn::extn_client_message::{ExtnMessage, ExtnResponse};
+
+const GATEWAY_DEFAULT: &str = "ws://127.0.0.1:9998/jsonrpc";
+
+#[derive(Deserialize, Clone)]
+pub struct ThunderPlatformParams {
+    #[serde(default = "gateway_default")]
+    gateway: String,
+}
+
+fn gateway_default() -> String {
+    String::from(GATEWAY_DEFAULT)
+}
 
 pub async fn boot_thunder(
-    state: ExtnClient,
+    ext_client: ExtnClient,
     plugin_param: ThunderPluginBootParam,
 ) -> Option<ThunderBootstrapStateWithClient> {
     info!("Booting thunder");
-    if let Ok(state) = ThunderGetConfigStep::setup(state, plugin_param).await {
-        if let Ok(state) = ThunderPoolStep::setup(state).await {
-            SetupThunderProcessor::setup(state.clone()).await;
-            return Some(state);
-        } else {
-            error!("Unable to connect to Thunder, error in ThunderPoolStep");
+    //by default enabling the thunderBroker
+    if ext_client.get_bool_config("use_with_thunder_broker") {
+        info!("Using thunder broker");
+        let mut extn_client = ext_client.clone();
+        let mut gateway_url = url::Url::parse(GATEWAY_DEFAULT).unwrap();
```

Comment:
Is this always going to use the default even if there is a value in the manifest? I think you want to pull from config and let the config logic decide whether or not to give you the default.